fake/source.go

```diff
@@ -0,0 +1,58 @@
+package fake
+
+import (
+	"io"
+	"math"
+	"sync"
+	"sync/atomic"
+)
+
+// Source is a pdk.Source which generates fake Event data.
+type Source struct {
+	max    uint64
+	events chan *Event
+	wg     sync.WaitGroup
+	closed chan struct{}
+	n      *uint64
+}
+
+// NewSource creates a new Source with the given random seed. Using the same
+// seed should give the same series of events on a given version of Go.
+func NewSource(seed int64, concurrency int, max uint64) *Source {
+	if max == 0 {
+		max = math.MaxUint64
+	}
+	var n uint64
+	s := &Source{
+		events: make(chan *Event, 1000),
+		closed: make(chan struct{}),
+		n:      &n,
+		max:    max,
+	}
+	for i := 0; i < concurrency; i++ {
+		s.wg.Add(1)
+		go func(i int) {
+			defer s.wg.Done()
+			g := NewEventGenerator(seed * 10 * int64(i))
```

Comment:
it's arbitrary, but the event generator actually uses seed and seed+1 because it uses math/rand directly and calls gen.NewGenerator which uses rand internally. I wanted to try to avoid seeds being re-used even if the implementation changed in the future. I will add a comment.